client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -0,0 +1,498 @@
+use {
+    crate::{
+        connection_cache::ConnectionCache,
+        nonblocking::{rpc_client::RpcClient, tpu_client::TpuClient},
+        rpc_client::RpcClient as BlockingRpcClient,
+    },
+    bincode::serialize,
+    dashmap::DashMap,
+    solana_quic_client::{QuicConfig, QuicConnectionManager, QuicPool},
+    solana_rpc_client::spinner,
+    solana_rpc_client_api::{
+        client_error::ErrorKind,
+        request::{RpcError, RpcResponseErrorData, MAX_GET_SIGNATURE_STATUSES_QUERY_ITEMS},
+    },
+    solana_sdk::{
+        hash::Hash,
+        message::Message,
+        signature::Signature,
+        signers::Signers,
+        transaction::{Transaction, TransactionError},
+    },
+    solana_tpu_client::{
+        nonblocking::tpu_client::set_message_for_confirmed_transactions,
+        tpu_client::{Result, TpuClientConfig, TpuSenderError},
+    },
+    std::{
+        sync::{
+            atomic::{AtomicU32, AtomicU64, Ordering},
+            Arc,
+        },
+        time::Duration,
+    },
+    tokio::{sync::RwLock, task::JoinHandle, time::Instant},
+};
+
+const BLOCKHASH_REFRESH_RATE: Duration = Duration::from_secs(10);
```

Comment:
I think there should be some constant saying for long the blockhash is valid and based on that we can set up refresh rate

Reply:
blockhashes aren't valid for a static period of time, but a fixed 150 blocks.  this can be biased in walltime by forking and skipped slots

Reply:
I think 5s will be a good refresh rate. Here the idea is that we do not unnecessarily poll the RPC, and should reflect a percentage of total expiry time.

So we have 300 blocks, considering 0 skips and 0 forks we have 300 slots = 120s.
Considering the average time to send and get transactions confirmed around 3s, 5s refresh time is a good compromise for everything.

Reply:
Maybe 5s is good enough, I just didn't get why 300 blocks are important and how 120s are used to compute these 5s

Reply:
it's _150_ blocks, because that's the `MAX_PROCESSING_AGE` of a recent blockhash. that is, any tx that references an rbh older than 150 blocks from the one considering including it is considered "expired" and non-executable.  120s presumably came from 300 blocks w/ skip-rate=0 -> 300 slots * 400ms target slot time. estimate, 120s. however the 300 should be a 150, so that estimate should be 60s. either way it's irrelevant because the 5s is arbitrary and only used to rate limit polling rpc for a new recent blockhash